userProfile/src/components/custom-modal/custom-modal.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Button, Modal } from '@skbkontur/react-ui';
+
+type CustomModalProps = {
+    message: string[];
+    onClose: () => void;
+};
+
+const CustomModal = ({ message, onClose }: CustomModalProps) => {

```

Comment:
Перенесли получение сообщений об изменениях сюда и считаем его налету (сразу реакт нодами, а почему бы и нет)
```suggestion
const CustomModal = ({ onClose, formData }: CustomModalProps) => {
    const getModalMessages = (state: FormState, prevState: null | FormState) => {
        const result: React.ReactNode[] = [];

        if (!prevState) {
            return [];
        }

        for (const key of Object.keys(state) as (keyof FormState)[]) {
            if (state[key] !== prevState[key]) {
                const message = `${FormDataNames[key]}: было ${prevState[key] || `не задано`}, стало ${state[key]}\n`;
                const messageNode = <p key={key}>{message}</p>;
                result.push(messageNode);
            }
        }

        return result;
    };
```